src/codegate/pipeline/base.py

```diff
@@ -260,19 +261,25 @@ def get_last_user_message_block(
         messages = request["messages"]
         block_start_index = None
 
+        base_tool = get_tool_name_from_messages(request)
+        accepted_roles = ["user", "assistant"]
+        if base_tool == "open interpreter":
+            # open interpreter also uses the role "tool"
+            accepted_roles.append("tool")
+
         # Iterate in reverse to find the last block of consecutive 'user' messages
         for i in reversed(range(len(messages))):
-            if messages[i]["role"] == "user" or messages[i]["role"] == "assistant":
-                content_str = messages[i].get("content")
+            if messages[i]["role"] in accepted_roles:
+                content_str = messages[i].get("content")                
                 if content_str is None:
                     continue
 
-                if messages[i]["role"] == "user":
+                if messages[i]["role"] in ["user", "tool"]:
                     user_messages.append(content_str)
                     block_start_index = i
 
                 # Specifically for Aider, when "Ok." block is found, stop
-                if content_str == "Ok." and messages[i]["role"] == "assistant":
+                if base_tool == "aider" and content_str == "Ok." and messages[i]["role"] == "assistant":
```

Comment:
how do we detect aider? I looked into get_tool_name_from_messages and I don't see aider returned from there

Reply:
i actually removed that, it is outdated, i removed the check for aider